projects/m5/002-polygon-area-calculator/js/src/index.js

```diff
@@ -0,0 +1,188 @@
+// @ts-check
+/**
+ * @fileoverview Create a Polygon Class.
+ * @author Salvatore Quagliariello
+ */
+
+
+/**
+ * @class Rectangle
+ * @classdesc A Class to create Rectangle Objects.
+ */
+class Rectangle {
+    /** @member {number} */
+    #width;
+    /** @member {number} */
+    #height;
+    /**
+     * @param {number} width - The width of the Rectangle Object.
+     * @param {number} height - The height of the Rectangle Object.
+     */
+    constructor(width, height) {
+        /** @private @property {number} */
+        this.#width = width;
+        /** @private @property {number} */
+        this.#height = height;
+    }
+
+    /** 
+     * @readonly
+     * @type {number} 
+     */
+    get width() {
+        return this.#width;
+    }
+
+    /** 
+     * @readonly
+     * @type {number} 
+     */
+    get height() {
+        return this.#height;
+    }
+
+    /**
+     * Set the value of the width of the Rectangle Object. 
+     * @param {number} value 
+     */
+    set_width(value) {
+        this.#width = value;
+    }
+
+    /**
+     * Set the value of the height of the Rectangle Object. 
+     * @param {number} value 
+     */
+    set_height(value) {
+        this.#height = value;
+    }
+
+    /**
+     * Returns the area of the Rectangle Object.
+     * @returns {number} area
+     */
+    get_area() {
+        return this.#width * this.#height;
+    }
+
+    /**
+     * Returns the perimeter of the Rectangle Object.
+     * @returns {number} perimeter
+     */
+    get_perimeter() {
+        return ((2 * this.#width) + (2 * this.#height));
+    }
+
+    /**
+     * Returns the diagonal of the Rectangle Object.
+     * @returns {number} diagonal
+     */
+    get_diagonal() {
+        return Math.sqrt((this.#width ** 2) + (this.#height ** 2));
+    }
+
+    /**
+     * Check if the `height` or the `width` of the Rectangle object are greater than 50.
+     * If so returns an error message. Otherwise returns a visual representation of the Object.
+     * @returns {string} A visual representation of the Object made out of `#` characters.
+     */
+    get_picture() {
+        if (this.#height > 50 || this.#width > 50) return "Too big for picture."
+
+        const ast = `*`;
+        let outputString = ``;
+
+        for (let i = 0; i < this.#height; i++) {
+            for (let k = 0; k < this.#width; k++) {
+                outputString += ast;
+            };
+            outputString += `\n`;
+        };
```

Comment:
Optiona Tip:
A possible alternative/linear solution could be:
- create width ("*" for **width**)
- repeat the row as many times as the height

What do you think about it?